File: convert_data.py
import json
import re

input_file = 'data.json'
output_file = 'data.json'

def parse_line(line):
    # Split by tab
    parts = line.strip().split('\t')
    # Remove leading + if present (first line had +=)
    # Remove =" and ending "
    cleaned = []
    for p in parts:
        # Regex to extract content between =" and "
        # Typical format: ="Some Text"
        # Sometimes: +="Some Text"
        match = re.search(r'="([^"]*)"', p)
        if match:
            cleaned.append(match.group(1))
        else:
            cleaned.append(p)
    return cleaned

try:
    with open(input_file, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    headers = parse_line(lines[0])
    data = []
    
    for line in lines[1:]:
        if not line.strip(): continue
        values = parse_line(line)
        
        # Create dict
        row = {}
        for i, h in enumerate(headers):
            val = values[i] if i < len(values) else ""
            row[h] = val
        data.append(row)
        
    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=4, ensure_ascii=False)
        
    print("Conversion successful. First item:")
    print(data[0])

except Exception as e:
    print(f"Error: {e}")

File: convert_data.js
const fs = require('fs');

const inputFile = 'data.json';
const outputFile = 'data.json';

try {
    const content = fs.readFileSync(inputFile, 'utf8');
    const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');

    // Parse line function: split by tab, remove +=" or =" wrappers
    const parseLine = (line) => {
        // Split by tab or assuming simple structure if tabs are lost (but pasted content usually keeps tabs)
        // From the view_file output, it looks like `="..."<tab>="..."`
        // We can split by `\t` first.
        const parts = line.split('\t');
        return parts.map(p => {
            // Match content inside "..."
            const match = p.match(/="([^"]*)"/);
            return match ? match[1] : p.replace(/^\+=?"/, '').replace(/"$/, '');
        });
    };

    const headers = parseLine(lines[0]);
    console.log("Headers:", headers);

    const data = [];
    for (let i = 1; i < lines.length; i++) {
        const values = parseLine(lines[i]);
        const row = {};
        headers.forEach((h, index) => {
            row[h] = values[index] || "";
        });
        data.push(row);
    }

    fs.writeFileSync(outputFile, JSON.stringify(data, null, 4), 'utf8');
    console.log("Conversion successful. Total records:", data.length);
    console.log("First item:", data[0]);

} catch (e) {
    console.error("Error:", e);
}
